sale_loyalty_limit/models/sale_order.py

```diff
@@ -0,0 +1,126 @@
+from odoo import _, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _check_all_max_customer_application(self, program):
+        """
+        Check if all rules associated with a loyalty program have the
+        'max_customer_application' property set to True.
+
+        :param program: Loyalty program object
+        :return: True if all rules meet the condition, False otherwise
+        """
+        return bool(all(rule.max_customer_application for rule in program.rule_ids))
```

Comment:
`all` already returns a Boolan value, and you can use mapped instead of a for. Anyway... do we need these hook methods for these simple one-liners?
```suggestion
        return all(program.rule_ids.mapped("max_customer_application"))
```

Reply:
They are not necessary when making the change at the program level, I had forgotten to delete them.